db/db_impl/db_impl_files.cc

```diff
@@ -100,6 +100,12 @@ Status DBImpl::EnableFileDeletions(bool force) {
   return Status::OK();
 }
 
+int DBImpl::EnableFileDeletionsWithLock(int count) {
+  mutex_.AssertHeld();
+  disable_delete_obsolete_files_ -= count;
```

Comment:
Should we set it to `std::max(0, disable_delete_obsolete_files_ - count)` in case somebody calls `EnableFileDeletions(true /* force */)`?

Reply:
Great catch!